tests/instruments/agilent/test_agilent33500.py

```diff
@@ -0,0 +1,154 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.agilent.agilent33500 import Agilent33500
+
+
+# TODO: Tests with a range in floats, seems to be flaky. Need to
+#       figure out the reason for protocol returning 6 after decimal point
+
+@pytest.mark.parametrize(
+    "shape", ["SIN", "SQU", "TRI", "RAMP", "PULS", "PRBS", "NOIS", "ARB", "DC"]
+)
+def test_shape(shape):
+    """
+    Test Agilent 33500 shape function
+    """
+    with expected_protocol(
+        Agilent33500,
+        [
+            ("SOUR1:FUNC?", shape),
+            ("SOUR2:FUNC?", shape),
+            ("FUNC?", shape),
+            (f"SOUR1:FUNC {shape}", None),
+            (f"SOUR2:FUNC {shape}", None),
+            (f"FUNC {shape}", None),
+        ],
+    ) as inst:
+        assert shape == inst.ch[1].shape
+        assert shape == inst.ch[2].shape
+        assert shape == inst.shape
+        inst.ch[1].shape = shape
+        inst.ch[2].shape = shape
+        inst.shape = shape
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("frequency", [1e-6, 1.2e8])
+def test_frequency(frequency):
+    """
+    Test Agilent 33500 frequency function
+    """
+    with expected_protocol(
+        Agilent33500,
+        [
+            ("SOUR1:FREQ?", frequency),
+            ("SOUR2:FREQ?", frequency),
+            ("FREQ?", frequency),
+            (f"SOUR1:FREQ {frequency}", None),
+            (f"SOUR2:FREQ {frequency}", None),
+            (f"FREQ {frequency}", None),
+        ],
+    ) as inst:
+        assert frequency == inst.ch[1].frequency
+        assert frequency == inst.ch[2].frequency
+        assert frequency == inst.frequency
+        inst.ch[1].frequency = frequency
+        inst.ch[2].frequency = frequency
+        inst.frequency = frequency
+
+
+@pytest.mark.xfail
```

Comment:
Why do you expect the tests to fail?

Please state the reason (in parenthesis) or make the test succeed.

Reply:
Removed xfail. I formatted the expected floats